sympy/assumptions/refine.py

```diff
@@ -1,6 +1,7 @@
 from typing import Dict, Callable
 
-from sympy.core import S, Add, Expr, Basic, Mul
+from sympy.core import S, Add, Expr, Basic, Mul, Pow, Rational
```

Comment:
Is it safe to import from the core package? Or should there be imports from their respective modules e.g.:
```python
from sympy.core.add import Add
from sympy.core.mul import Mul
from sympy.core.power import Pow
```

Reply:
Ideally from the modules, but I only wanted to spend so much time (sometimes I changed it to modules, but not everywhere).

Reply:
But it is in general safe. Makes it clearer though to import from the modules.